### ISSUE TITLE
ActiveRecord::Generators::GeneratedAttribute does not set the :type option to :uuid on references when primary_key_type is :uuid

### ISSUE THREAD

**Issue by Korkey128k** [rails/rails#40010]
### Steps to reproduce
Run
```
rails new app -d postgresql --skip-spring --skip-javascript && \
cd app && \
rails g migration enable_uuid
```

Add to `config/initializers/generators.rb`
```
Rails.application.config.generators do |g|
  g.orm :active_record, primary_key_type: :uuid
end
```

Modify `db/migrations/{datestamp}_enable_uuid.rb` to:
```
class EnableUuid < ActiveRecord::Migration[6.0]
  def change
    enable_extension 'pgcrypto' unless extension_enabled?('pgcrypto')
  end
end
```

Run
```
rails g model Post title:string && \
rails g model Tag post:references && \
rails db:create db:migrate
```

### Expected behavior
I had assumed that stepping through these migrations would have created the correct `references` attribute migration needed for any references to `:uuid` primary key records.  
Instead, it produces an error (shown below)

### Actual behavior
A `rails db:create db:migrate` throws an error:
```
PG::DatatypeMismatch: ERROR:  foreign key constraint "fk_rails_d1d3ea39b4" cannot be implemented
DETAIL:  Key columns "post_id" and "id" are of incompatible types: bigint and uuid.
```

### System configuration
**Rails version**: `6.0.3.2`

**Ruby version**: `2.7.1p83`

---

### Additional Details
Wasn't sure how to make a bug report for this issue, otherwise I would have. (How do you run generators in a bug report?)

~Also, during my sleuthing, I realized [the documentation for this](https://edgeguides.rubyonrails.org/active_record_postgresql.html#uuid) is out of date as well. (I don't need the `default: 'gen_random_uuid()'` line on `create_table :devices`.~ (Fixed in #40012) ~Furthermore, it doesn't specify anything about adding `references type: :uuid` to a migration of any relations.~ (This was just plain false, it did)

**Comment by Korkey128k** [comment]
Pull Request #40012 for documentation fix merged.

**Comment by eugeneius** [comment]
Thank you for the report, but this has already been fixed in https://github.com/rails/rails/pull/37583 which will be released in Rails 6.1.
